You-Can-Code-It/fndr
# Pagination feature

# What this pull request does

## What

https://github.com/luizfiorentino/fndr/assets/96445830/c749aba5-1bc5-430e-b01b-437f399ef8a0


- adds pagination to the list of displayed company cards on the index page
- the company cards are displayed in page groups (starts from 1-5, then 6-10 etc.)
- there are  the "Back" and "Next" buttons to navigate backwards or forwards between the page groups

## Why

- many companies are displayed in the index page, so a pagination feature is welcome to display a small number of them at each time
- since the number of generated buttons is also too big, it is necessary to display them in groups using the "Back" and "Next" buttons

## How

- a "Pagination" function is created in a "serialize.js" file to dynamically calculate the total number of buttons from the companies data, specify a number of cards (8) and group of buttons (5) displayed at each time, and generate the correspondent page group and data result for the user
- the "Back" and "Next" buttons lead to the previous or next 5 results (starts on 1-5, then 6-10 and so on)
- this function is imported and used in the index page that displays 8 companies at a time
- when the user clicks at a numbered number, the correspondent group of 8 company cards is shown on screen (starts from the first group of it)
- clicking the "Back" and "Next" buttons lead the user to the previous or next page group
- the buttons bar is responsively styled for different screen sizes

## Steppings to test this

- go to the index page, and check out the buttons bar
- click on each number and the displayed results shall change
- test also the "Back" and "Next" buttons, if they show the previous or next page group 

## Anything else

- maybe add the "First" and "Last" buttons to the  pagination buttons bar, since there is a huge number of companies and it provides the user with an accurate idea of the amount of data
- this feature could also be complemented with other ones like searching all companies starting with a particular letter (e.g. using a filter)
